modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/common/util/UsernameTokenValidator.java

```diff
@@ -0,0 +1,131 @@
+import org.wso2.carbon.user.api.UserRealm;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.service.RealmService;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+/**
+ * Validator class for user authentication checking the default carbon user store.
+ */
+public class UsernameTokenValidator implements Validator {
+
+	private static final int USER_SEGMENT = 0;
+	private static final int DOMAIN_SEGMENT = 1;
+	private static final String DELIMITER = "@";
+	private static Log logger = LogFactory.getLog(UsernameTokenValidator.class);
+
+	/**
+	 * This method validates the username token in SOAP message coming from the device.
+	 *
+	 * @param credential - Username token credentials coming from device
+	 * @param requestData - Request data associated with the request
+	 * @return - Credential object if authentication is success, or null if not success
+	 * @throws WSSecurityException
+	 */
+	@Override
+	public Credential validate(Credential credential, RequestData requestData)
+			throws WSSecurityException {
+
+		String domainUser = credential.getUsernametoken().getName();
+		String[] domainUserArray = domainUser.split(DELIMITER);
+		Credential returnCredentials;
+		String user = domainUserArray[USER_SEGMENT];
+		String domain = domainUserArray[DOMAIN_SEGMENT];
+		String password = credential.getUsernametoken().getPassword();
+
+		try {
+			if (authenticate(user, password, domain)) {
+				returnCredentials = credential;
+			} else {
+				String msg="Authentication failure due to incorrect credentials.";
+				logger.error(msg);
+				throw new WindowsDeviceEnrolmentException(msg);
+			}
+		//Generic exception is caught here as there is no need of taking different actions for
+		//different exceptions.
+		} catch (Exception e) {
+			String msg = "Failure occurred in the credential validator.";
+			logger.error(msg);
+			throw new WSSecurityException(msg);
+		}
+		return returnCredentials;
+	}
+
+	/**
+	 * This method authenticate the user checking the carbon default user store.
+	 *
+	 * @param username - Username in username token
+	 * @param password - Password in username token
+	 * @param tenantDomain - Tenant domain is extracted from the username
+	 * @return - Returns boolean representing authentication result
+	 * @throws AuthenticationException
+	 */
+	public boolean authenticate(String username, String password, String tenantDomain)
+			throws AuthenticationException {
+
+		try {
+			PrivilegedCarbonContext.startTenantFlow();
+			PrivilegedCarbonContext ctx = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+			ctx.setTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+			ctx.setTenantId(MultitenantConstants.SUPER_TENANT_ID);
+			RealmService realmService = (RealmService) ctx.getOSGiService(RealmService.class, null);
+
+			if (realmService == null) {
+				String msg = "RealmService not initialized.";
```

Comment:
As it appears, there seem to be more than 4 spaces in the indentation template being used. Please configure the IDE properly to avoid similar instances.